examples/infra/modDA.rs

```diff
@@ -0,0 +1,775 @@
+        let mut total_transactions = 0;
+        let mut total_commitments = 0;
+
+        let start = Instant::now();
+
+        error!("Starting hotshot!");
+        let (mut event_stream, _streamid) = context.get_event_stream(FilterEvent::default()).await;
+        let mut anchor_view: TYPES::Time = <TYPES::Time as ConsensusTime>::genesis();
+        let mut num_successful_commits = 0;
+
+        let total_nodes_u64 = total_nodes.get() as u64;
+
+        let mut should_submit_txns = node_index == (*anchor_view % total_nodes_u64);
+
+        context.hotshot.start_consensus().await;
+
+        loop {
+            if should_submit_txns {
+                for _ in 0..transactions_per_round {
+                    let txn = txns.pop_front().unwrap();
+                    tracing::error!("Submitting txn on round {}", round);
+                    context.submit_transaction(txn).await.unwrap();
+                    total_transactions += 1;
+                }
+                should_submit_txns = false;
+            }
+
+            match event_stream.next().await {
+                None => {
+                    panic!("Error! Event stream completed before consensus ended.");
+                }
+                Some(Event { view_number, event }) => {
+                    match event {
+                        EventType::Error { error } => {
+                            error!("Error in consensus: {:?}", error);
+                            // TODO what to do here
+                        }
+                        EventType::Decide { leaf_chain, qc } => {
+                            error!("In decide handler");
+                            // this might be a obob
+                            if let Some(leaf) = leaf_chain.get(0) {
+                                error!("Decide event for leaf: {}", *leaf.view_number);
+
+                                let new_anchor = leaf.view_number;
+                                if new_anchor >= anchor_view {
+                                    anchor_view = leaf.view_number;
+                                }
+                                if (*anchor_view % total_nodes_u64) == node_index {
+                                    should_submit_txns = true;
+                                }
+                            }
+                            num_successful_commits += leaf_chain.len();
+                            if num_successful_commits >= rounds {
+                                break;
+                            }
+                            // when we make progress, submit new events
+                        }
+                        EventType::ReplicaViewTimeout { view_number } => {
+                            error!("Timed out as a replicas in view {:?}", view_number);
+                        }
+                        EventType::NextLeaderViewTimeout { view_number } => {
+                            error!("Timed out as the next leader in view {:?}", view_number);
```

Comment:
We should be able to drop this part too if we integrate the timeout task (or a configured safety task, which tracks successful views).